packages/streaming-image-volume-loader/src/BaseStreamingImageVolume.ts

```diff
@@ -698,37 +727,36 @@ export default class BaseStreamingImageVolume extends ImageVolume {
     throw new Error('Abstract method');
   }
 
-  private _prefetchImageIds(priority: number): void {
+  public getImageIdsLoad(): string[] {
+    throw new Error('Abstract method');
+  }
+
+  private _prefetchImageIds(): Promise<unknown> {
     // Note: here is the correct location to set the loading flag
     // since getImageIdsRequest is just grabbing and building requests
     // and not actually executing them
     this.loadStatus.loading = true;
 
-    const requests = this.getImageLoadRequests(priority);
+    const imageIds = this.getImageIdsLoad();
+    this.numFrames = imageIds.length;
 
-    requests.reverse().forEach((request) => {
-      if (!request) {
-        // there is a cached image for the imageId and no requests will fire
-        return;
-      }
+    this.totalNumFrames = this.imageIds.length;
+    const autoRenderPercentage = 2;
 
-      const {
-        callLoadImage,
-        imageId,
-        imageIdIndex,
-        options,
-        priority,
-        requestType,
-        additionalDetails,
-      } = request;
+    if (this.autoRenderOnLoad) {
+      this.reRenderFraction =
+        this.totalNumFrames * (autoRenderPercentage / 100);
+      this.reRenderTarget = this.reRenderFraction;
+    }
 
-      imageLoadPoolManager.addRequest(
-        callLoadImage.bind(this, imageId, imageIdIndex, options),
-        requestType,
-        additionalDetails,
-        priority
-      );
-    });
+    return progressiveLoader
+      .load(imageIds, this)
```

Comment:
we shouldn't assume the following in the progressiveLoader

```js
    if (options.target?.arrayBuffer) {
      loadedPromise = loadAndCacheImage(imageId, options);
    } else {
      loadedPromise = loadImage(imageId, options);
    }
```

ideally we should just have 

`progressiveLoader.loadImage` or `progressiveLoader.loadAndCacheImage` and internally we handle the argument passing to your current load